nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationService.java

```diff
@@ -459,7 +460,58 @@ public void testDeleteExternalAnnotation() {
         assertFalse(session.exists(new IdRef(annotation.getId())));
 
     }
-    
+
+    @Test
+    public void testGetTopLevelAnnotationAncestor() {
+        DocumentModel docToAnnotate = session.createDocumentModel("/testDomain", "testDoc", "File");
+        docToAnnotate = session.createDocument(docToAnnotate);
+
+        String entityId = "foo";
+        String docIdToAnnotate = docToAnnotate.getId();
+        String xpathToAnnotate = "files:files/0/file";
+        String comment = "test comment";
+        String origin = "Test";
+        String entity = "<entity><annotation>bar</annotation></entity>";
+
+        Annotation annotation = new AnnotationImpl();
+        annotation.setAuthor("jdoe");
+        annotation.setText(comment);
+        annotation.setParentId(docIdToAnnotate);
+        annotation.setXpath(xpathToAnnotate);
+        annotation.setCreationDate(Instant.now());
+        annotation.setModificationDate(Instant.now());
+        ((ExternalEntity) annotation).setEntityId(entityId);
+        ((ExternalEntity) annotation).setOrigin(origin);
+        ((ExternalEntity) annotation).setEntity(entity);
+        annotation = annotationService.createAnnotation(session, annotation);
+
+        ACP acp = docToAnnotate.getACP();
+        ACL acl = acp.getOrCreateACL();
+        acl.add(new ACE("james", SecurityConstants.READ, true));
+        session.setACP(docToAnnotate.getRef(), acp, false);
+        session.save();
+
+        CommentManager commentManager = Framework.getService(CommentManager.class);
+        try (CloseableCoreSession systemSession = coreFeature.openCoreSession()) {
+            assertEquals(docToAnnotate.getRef(),
+                    commentManager.getTopLevelCommentAncestor(systemSession, new IdRef(annotation.getId())));
+        }
+
+        try (CloseableCoreSession jamesSession = coreFeature.openCoreSession("james")) {
+            assertEquals(docToAnnotate.getRef(),
+                    commentManager.getTopLevelCommentAncestor(jamesSession, new IdRef(annotation.getId())));
+        }
+
+        try (CloseableCoreSession janeSession = coreFeature.openCoreSession("jane")) {
+            assertEquals(docToAnnotate.getRef(),
+                    commentManager.getTopLevelCommentAncestor(janeSession, new IdRef(annotation.getId())));
+        } catch (CommentSecurityException cse) {
```

Comment:
There should be a `fail()` somewhere if you expect the `catch` to be triggered, otherwise you're not testing anything.

Reply:
yep, i just forgot it :) thx, fyi i will add more test to cover the case of get top level on comment